lib/versionmanager.js

```diff
@@ -231,7 +231,24 @@ async function upgradePackageDefinitions(currentDependencies, options) {
     return !options.jsonUpgraded || !options.minimal || !isSatisfied(latestVersions[dep], currentDependencies[dep])
   })
 
-  return [filteredUpgradedDependencies, latestVersions]
+  let result = [filteredUpgradedDependencies, latestVersions, options.peerDependencies]
+  if (options.peer && !_.isEmpty(filteredUpgradedDependencies)) {
+    const upgradedPeerDependencies = await getPeerDependenciesFromRegistry(filteredUpgradedDependencies, options)
+    const peerDependencies = { ...options.peerDependencies, ...upgradedPeerDependencies }
+    result[2] = peerDependencies
```

Comment:
If you could find a way to do this without mutations that would be great. They introduce ordering constraints into the code and I consider them a bad practice. For example, in this code it is less clear than it could be all of the places that `result` is set. With a more functional approach, the code structure itself would help to show the code paths that lead to different `result` values.

Reply:
Thanks for your feedback! I've fixed all issues except this one. I will have to think about, how I can improve this part. But I agree, that the readability can be improved :smiley:

Reply:
I added a fix for this (the `result[2] = peerDependencies` wasn't even necessary :roll_eyes: ). Is it better, now?

Reply:
Yes, thank you!